### ISSUE TITLE
Crash in Android gradle 4.1.1

### ISSUE THREAD

**Issue by newbision** [koush/ion#944]
I upgraded the build Gradle to 4.1.1 and the app is crashed. Please check the following log.

```
Fatal Exception: java.lang.AssertionError
       at com.koushikdutta.async.AsyncSSLSocketWrapper.write(AsyncSSLSocketWrapper.java:492)
       at com.koushikdutta.async.AsyncSSLSocketWrapper.handleHandshakeStatus(AsyncSSLSocketWrapper.java:378)
       at com.koushikdutta.async.AsyncSSLSocketWrapper.handshake(AsyncSSLSocketWrapper.java:182)
       at com.koushikdutta.async.http.AsyncSSLSocketMiddleware.tryHandshake(AsyncSSLSocketMiddleware.java:89)
       at com.koushikdutta.async.http.AsyncSSLSocketMiddleware$2.onConnectCompleted(AsyncSSLSocketMiddleware.java:106)
       at com.koushikdutta.async.http.AsyncSocketMiddleware.lambda$getSocket$3$AsyncSocketMiddleware(AsyncSocketMiddleware.java:217)
       at com.koushikdutta.async.http.-$$Lambda$AsyncSocketMiddleware$glazjOxu-iNIRY4N890Wl1ngyGo.onCompleted(:12)
       at com.koushikdutta.async.future.SimpleFuture.lambda$setCallback$0(SimpleFuture.java:212)
       at com.koushikdutta.async.future.-$$Lambda$SimpleFuture$h_gK5Y1XHZI1H8JTE2xOg4dXbp8.onCompleted(:2)
       at com.koushikdutta.async.future.SimpleFuture$FutureCallsite.loop(SimpleFuture.java:130)
       at com.koushikdutta.async.future.SimpleFuture.handleCallbackUnlocked(SimpleFuture.java:153)
       at com.koushikdutta.async.future.SimpleFuture.setComplete(SimpleFuture.java:191)
       at com.koushikdutta.async.future.SimpleFuture.setComplete(SimpleFuture.java:174)
       at com.koushikdutta.async.future.-$$Lambda$VNAuWcttAhMr3VsWCERbmWE6yR0.success(:2)
       at com.koushikdutta.async.future.SimpleFuture.lambda$success$4(SimpleFuture.java:279)
       at com.koushikdutta.async.future.-$$Lambda$SimpleFuture$DxIsaj7NQi76eWgOKS4iviWPLV0.onCompleted(:4)
       at com.koushikdutta.async.future.SimpleFuture$FutureCallsite.loop(SimpleFuture.java:130)
       at com.koushikdutta.async.future.SimpleFuture.handleCallbackUnlocked(SimpleFuture.java:153)
       at com.koushikdutta.async.future.SimpleFuture.setComplete(SimpleFuture.java:191)
       at com.koushikdutta.async.future.SimpleFuture.setComplete(SimpleFuture.java:178)
       at com.koushikdutta.async.http.-$$Lambda$cBC5du-82c5EcL72rP5wNK7Y2Y4.onConnectCompleted(:2)
       at com.koushikdutta.async.AsyncServer.runLoop(AsyncServer.java:896)
       at com.koushikdutta.async.AsyncServer.run(AsyncServer.java:717)
       at com.koushikdutta.async.AsyncServer.access$800(AsyncServer.java:46)
       at com.koushikdutta.async.AsyncServer$11.run(AsyncServer.java:671)
```

**Comment by Toolo** [comment]
See https://github.com/koush/ion/issues/937#issuecomment-723546242

This was fixed in v3.1.0

**Comment by newbision** [comment]
Thank you for your update. I will check it and let you know.

**Comment by newbision** [comment]
Great! It's fixed in v3.1.0. Would you please update a new package in the Github repository?
